io/io/src/TStreamerInfo.cxx

```diff
@@ -3481,11 +3481,8 @@ static void R__WriteMoveBodyPointersArrays(FILE *file, const TString &protoname,
             if (element->GetArrayDim() == 1) {
                fprintf(file,"   for (Int_t i=0;i<%d;i++) %s[i] = rhs.%s[i];\n",element->GetArrayLength(),ename,ename);
             } else if (element->GetArrayDim() >= 2) {
-               fprintf(file,"   for (Int_t i=0;i<%d;i++) (&(%s",element->GetArrayLength(),ename);
-               for (Int_t d = 0; d < element->GetArrayDim(); ++d) {
-                  fprintf(file,"[0]");
-               }
-               fprintf(file,"))[i] = (&(rhs.%s",ename);
+               fprintf(file,"   for (Int_t i=0;i<%d;i++) reinterpret_cast<%s *>(%s", element->GetArrayLength(), element->GetTypeName(), ename);
+               fprintf(file,")[i] = (&(rhs.%s",ename);
```

Comment:
Would it make sense to handle the rhs identically? Or does this trigger yet another warning?

Reply:
I'll try that.

Reply:
It works fine.  Thanks.